[From http://mirror.keystealth.org/gnu/bash/bash-5.1-patches/bash51-002]
Signed-off-by: Peter Seiderer <ps.report@gmx.net>

			     BASH PATCH REPORT
			     =================

Bash-Release:	5.1
Patch-ID:	bash51-002

Bug-Reported-by:	oguzismailuysal@gmail.com
Bug-Reference-ID:	<CAH7i3LoHFUa4aSF5-AD2r80HG-p-YzD_9ZxomarZkhP8NMq63g@mail.gmail.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2020-12/msg00037.html

Bug-Description:

If there are no jobs, and the `-n' and `-p' options are both supplied to
`wait', bash can assign a value to the variable name specified with `-p'
instead of leaving it unset.

Patch (apply with `patch -p0'):

*** ../bash-5.1-patched/builtins/wait.def	2020-04-09 15:13:57.000000000 -0400
--- b/builtins/wait.def	2020-12-11 09:46:49.000000000 -0500
***************
*** 214,222 ****
  
        status = wait_for_any_job (wflags, &pstat);
-       if (status < 0)
- 	status = 127;
- 
        if (vname && status >= 0)
  	bind_var_to_int (vname, pstat.pid);
        if (list)
  	unset_waitlist ();
--- 214,222 ----
  
        status = wait_for_any_job (wflags, &pstat);
        if (vname && status >= 0)
  	bind_var_to_int (vname, pstat.pid);
+ 
+       if (status < 0)
+ 	status = 127;
        if (list)
  	unset_waitlist ();

*** ../bash-5.1/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- b/patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 1
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 2
  
  #endif /* _PATCHLEVEL_H_ */
